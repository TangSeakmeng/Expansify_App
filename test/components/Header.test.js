import React from 'react';
import { shallow } from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';
import Header from '../../src/componenets/Header';

test('should render header correctly using shallow', () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();

    // expect(wrapper.find('h1').length).toBe(1);
    // expect(wrapper.find('h1').text()).toBe('Expansify App');
})

test('should render header correctly using ReactShallowRenderer', () => {
    const render = new ReactShallowRenderer();
    render.render(<Header/>);
    expect(render.getRenderOutput()).toMatchSnapshot();
})