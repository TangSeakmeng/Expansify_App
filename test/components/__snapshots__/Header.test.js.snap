// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly using ReactShallowRenderer 1`] = `
<header>
  <h1>
    Expansify App
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    help
  </NavLink>
</header>
`;

exports[`should render header correctly using shallow 1`] = `
<header>
  <h1>
    Expansify App
  </h1>
  <[object Object]
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Dashboard
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/create"
  >
    Create Expense
  </[object Object]>
  <[object Object]
    activeClassName="is-active"
    to="/help"
  >
    help
  </[object Object]>
</header>
`;
